### ISSUE TITLE
Needs light mode

### ISSUE THREAD

**Issue by sreekaransrinath** [OurTechCommunity/website#9]
Dark background + light text is not very accessible/vision-impairment friendly. 

Imo we should have a light mode that people can at least shift to.

**Comment by tusharnankani** [comment]
@HarshKapadia2 what do you feel about it?

@sreekaransrinath, instead of adding a light mode, do you think softening the colors but retaining the mode would help?

Either way, we'll have to modify [css](https://github.com/OurTechCommunity/website/blob/main/static/css/style.css) in all repositories.

**Comment by HarshKapadia2** [comment]
I don't mind supporting a light mode, but you're right @tusharnankani, we'll have to change the CSS, add theme switchers, check responsiveness and all of that everywhere, which will be a task.

I keep thinking that we need to build some kind of a npm package or some plugin system to be able to update and keep our UI consistent across repos. I know that Razorpay does this. It will be time consuming, but it will reduce work going down the line, I guess. I've also wanted to include Web Components for repetitive things like our burger menus, because adding one page implies changing the menu HTML on every page, which is obviously irritating (https://github.com/OurTechCommunity/catchup/issues/55). (Cc: @KartikSoneji)

That being said though, I don't think we have the time to do any of these (add a separate theme or create a template) for now. Good suggestion though @sreekaransrinath. This will be an update that we will look to add in the future.

**Comment by KartikSoneji** [comment]
> Either way, we'll have to modify [css](https://github.com/OurTechCommunity/website/blob/main/static/css/style.css) in all repositories.

How about a page-wide invert filter for everthing except images?

**Comment by tusharnankani** [comment]
Yep, possible. 

```css
html {
    filter: invert(1) hue-rotate(180deg);
}

img {
    /* reverse the effect for images, svgs and other assets*/
    filter: invert(1) hue-rotate(180deg); 
}
```

**Comment by KartikSoneji** [comment]
I think we can use the [`:not()` selector](https://developer.mozilla.org/en-US/docs/Web/CSS/:no)

```css
:not(img, svg) {
    filter: invert(1) hue-rotate(180deg);
}
```

**Comment by tusharnankani** [comment]
Best.

**Comment by HarshKapadia2** [comment]
Good idea @KartikSoneji. Here is how it looks on our OTC home page. It seems okay to me for this page, but I haven't checked other pages.

Before:

![image](https://user-images.githubusercontent.com/50140864/198457794-27f2a8de-5533-4b92-aea7-78705f7cba0b.png)

After:

![image](https://user-images.githubusercontent.com/50140864/198457071-bbf5bf42-93b0-464e-90cd-e2bcb0345bb6.png)

**Comment by tusharnankani** [comment]
Too strong colours. That’s why I said — let’s soften the colours.

**Comment by HarshKapadia2** [comment]
How?

**Comment by tusharnankani** [comment]
I meant the colors are too strong, a little too contrasting.
And sometimes, it is difficult to read.

**Comment by HarshKapadia2** [comment]
Okay, but how do we do that without adding a lot of CSS across all our repos?

**Comment by KartikSoneji** [comment]
We can add a selector like:

```css
body.invert :not(img, svg) {
    filter: invert(1) hue-rotate(180deg);
}
```

Then have the theme selector toggle the class:
```js
document.body.classList.toggle("invert");
```

**Comment by HarshKapadia2** [comment]
Yes, but as [Tushar said](https://github.com/OurTechCommunity/website/issues/9#issuecomment-1296218290):

> I meant the colors are too strong, a little too contrasting.
> And sometimes, it is difficult to read.

**Comment by KartikSoneji** [comment]
`invert(0.97)` instead of `invert(1)` ?

Side note:
Maybe there can be a common css file across all repos that only defines css variables for the theme colors.

**Comment by HarshKapadia2** [comment]
> `invert(0.97)` instead of `invert(1)` ?

What do you think @tusharnankani?

> Side note:
> Maybe there can be a common css file across all repos that only defines css variables for the theme colors.

Why not have a common CSS library in that case?

**Comment by KartikSoneji** [comment]
> > Side note:
> > Maybe there can be a common css file across all repos that only defines css variables for the theme colors.
> 
> Why not have a common CSS library in that case?

A common library will require a full rewrite (which is not a bad thing, just not sure if we have the time for that)
A common file with css variables will be much easier to integrate.

There should be a tool/webpack plugin to do something like that I think.
If not, it should be easy enough to build one.

**Comment by HarshKapadia2** [comment]
> A common library will require a full rewrite (which is not a bad thing, just not sure if we have the time for that)
> A common file with css variables will be much easier to integrate.

Isn't it a temporary solution that might become permanent if we don't handle it properly? I agree that we don't have the time right now, but maybe we can start work on it and progress slowly? The light mode issue doesn't seem pressing, so maybe we can do it the correct way? I know Accessibility is important, so I'm not sure what we should do. What do you think @KartikSoneji?

Should we do both? Implement the temporary solution (which will also consume time, but less of it) and start work on the common library as well (which will consume a lot of time)?

> There should be a tool/webpack plugin to do something like that I think.
> If not, it should be easy enough to build one.

How would this thing work?

**Comment by KartikSoneji** [comment]
> > A common library will require a full rewrite (which is not a bad thing, just not sure if we have the time for that)
> > A common file with css variables will be much easier to integrate.
> 
> Isn't it a temporary solution that might become permanent if we don't handle it properly? I agree that we don't have the time right now, but maybe we can start work on it and progress slowly? The light mode issue doesn't seem pressing, so maybe we can do it the correct way? I know Accessibility is important, so I'm not sure what we should do. What do you think @KartikSoneji?
> 
> Should we do both? Implement the temporary solution (which will also consume time, but less of it) and start work on the common library as well (which will consume a lot of time)?

Sorry I should have been more clear.
When I say you should try to avoid temporary solutions that might become permanant, I mean temporary solutions that are not the ideal way of doing things.
Whenever we start building a common library we anyways need to switch all the colors to theme variables.
Fortunately, in this case the temporary solution lends itself well to gradual adoption. We can slowly switch out parts of the site with a common framework as and when they're done.

The `invert(1)` filter is exactly the type of patch I try to avoid, but in this case it gets us 95% of the way there with a negligble amount of work, and is trival to roll back, so I think it's worth the tradeoff.

> > There should be a tool/webpack plugin to do something like that I think.
> > If not, it should be easy enough to build one.
> 
> How would this thing work?

- Parse CSS files with babel
- List all hardcoded colors
- Replace hardcoded colors with CSS variables

**Comment by HarshKapadia2** [comment]
> > > There should be a tool/webpack plugin to do something like that I think.
> > > If not, it should be easy enough to build one.
> > 
> > How would this thing work?
> 
> - Parse CSS files with babel
> - List all hardcoded colors
> - Replace hardcoded colors with CSS variables

That feels hacky. Won't we have to scrap this when implementing the common library?

> Fortunately, in this case the temporary solution lends itself well to gradual adoption. We can slowly switch out parts of the site with a common framework as and when they're done.


How does the temporary solution lead to the final one?

**Comment by KartikSoneji** [comment]
> > > > There should be a tool/webpack plugin to do something like that I think.
> > > > If not, it should be easy enough to build one.
> > > 
> > > 
> > > How would this thing work?
> > 
> > 
> > 
> > * Parse CSS files with babel
> > * List all hardcoded colors
> > * Replace hardcoded colors with CSS variables
> 
> That feels hacky. Won't we have to scrap this when implementing the common library?

No, the idea is we replace all colors with css variables.

> > Fortunately, in this case the temporary solution lends itself well to gradual adoption. We can slowly switch out parts of the site with a common framework as and when they're done.
> 
> How does the temporary solution lead to the final one?

As the common framwork develops, more and more site-specific css can be removed.

**Comment by HarshKapadia2** [comment]
Okay. Please propose a plan for this, @KartikSoneji.

Also, is this the correct place to discuss this or should we move this to somewhere else? If not, please propose the plan at the other place. Maybe a new issue?

**Comment by SirusCodes** [comment]
@KartikSoneji IMO temporary solutions are the most permanent one.

Implementing light mode properly is what I feel is a better way to do it.

**Comment by HarshKapadia2** [comment]
As mentioned before, @SirusCodes:

> > > Fortunately, in this case the temporary solution lends itself well to gradual adoption. We can slowly switch out parts of the site with a common framework as and when they're done.
> > 
> > How does the temporary solution lead to the final one?
> 
> As the common framwork develops, more and more site-specific css can be removed.
